### PR TITLE
DEPR: Deprecate numpy argument in read_json

### LINKED ISSUE #28512: DEPR: Deprecate numpy argument in read_json
I've never really been clear on the purpose of the `numpy` argument in to_json. Some digging brought me here https://github.com/pandas-dev/pandas/pull/3876#issuecomment-19390383 where it is explained that this maintains some kind of sequence to elements. To illustrate the only difference I could find

```python
>>> pd.read_json('[{"a": 1, "b": 2}, {"b":2, "a" :1}]',numpy=False, orient='records')
   a  b
0  1  2
1  1  2

>>> pd.read_json('[{"a": 1, "b": 2}, {"b":2, "a" :1}]',numpy=True, orient='records')
   a  b
0  1  2
1  2  1
```

I might be missing the point but I don't understand why this would be useful. Objects in JSON are by definition not ordered, so this is non-compliant and I think just plain confusing.

So I think good to deprecate unless anyone has objections.

### PR DESCRIPTION
**Pull request by alimcmaster1**
Co-authored-by: Luca Ionescu <lucaionescu@users.noreply.github.com>

- Continuing https://github.com/pandas-dev/pandas/pull/28562
- [x] closes #28512

@lucaionescu - i've merged master and pushed to here (I don't have the permissions to push to your branch), I will aim to fix up the tests. Or feel free to take it from here if you have time?

Opening up WIP PR to see what test failures need addressing. 

I've also co-authored the commit.

### CODE REVIEW

File: pandas/tests/io/json/test_pandas.py [hunk 1]
```diff
@@ -1606,3 +1607,13 @@ def test_emca_262_nan_inf_support(self):
             ["a", np.nan, "NaN", np.inf, "Infinity", -np.inf, "-Infinity"]
         )
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.filterwarnings("ignore:.*msgpack:FutureWarning")
```
Comment (jreback) [inline pandas/tests/io/json/test_pandas.py]: msgpack?

File: pandas/tests/io/json/test_pandas.py [hunk 2]
```diff
@@ -1606,3 +1607,13 @@ def test_emca_262_nan_inf_support(self):
...
+    def test_deprecate_numpy_argument_read_json(self):
+        # https://github.com/pandas-dev/pandas/issues/28512
+        expected = DataFrame([1, 2, 3])
+        with tm.assert_produces_warning(None):
+            with catch_warnings():
```
Comment (jreback) [inline pandas/tests/io/json/test_pandas.py]: why are you catching warnings here?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/json/test_pandas.py [hunk 3]
```diff
@@ -1606,3 +1607,13 @@ def test_emca_262_nan_inf_support(self):
             ["a", np.nan, "NaN", np.inf, "Infinity", -np.inf, "-Infinity"]
         )
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.filterwarnings("ignore:.*msgpack:FutureWarning")
+    def test_deprecate_numpy_argument_read_json(self):
+        # https://github.com/pandas-dev/pandas/issues/28512
+        expected = DataFrame([1, 2, 3])
+        with tm.assert_produces_warning(None):
```
Comment (WillAyd) [inline pandas/tests/io/json/test_pandas.py]:
```suggestion
        with tm.assert_produces_warning(FutureWarning):
```

Should work and make things easier

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/io/json/test_pandas.py (hunk 2 above)
Comment (alimcmaster1) [reply pandas/tests/io/json/test_pandas.py]: Yep unsure why this was in https://github.com/pandas-dev/pandas/pull/28562. Now simplified

File: setup.cfg [hunk 4]
```diff
@@ -66,6 +66,7 @@ xfail_strict = True
 filterwarnings =
     error:Sparse:FutureWarning
     error:The SparseArray:FutureWarning
+    ignore:  the 'numpy' keyword is deprecated:FutureWarning
```
Comment (jreback) [inline setup.cfg]: don't do this, instead wrap the appropriate tests

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by WillAyd** [comment]
Can you merge master?

**Comment by alimcmaster1** [comment]
> Can you merge master?

done!

**Review by WillAyd — APPROVED** [review]
lgtm @jreback

**Comment by alimcmaster1** [comment]
friendly ping @jreback

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @alimcmaster1 very nice!

**Pull request merged**